west/tools.py

```diff
@@ -176,5 +176,6 @@ def monitor(self, module_path, args):
             cmd_exec(("python.exe", monitor_path, "-p", esp_port,
                      "-b", args.baud, elf_path, "--eol", args.eol), cwd=cmd_path)
         else:
+            cmd_path = Path(os.getcwd())
```

Comment:
1) Does that work if you build an external project within zephyr folder?   
`<zephyrproject/zephyr>west build -b esp32 </home/xx/samples/heello_world> `

Reply:
I've tested building in-tree and out-tree samples and it build & flash & run with monitor as expected. But I'd like anyone else to test it on their machines too.